### ISSUE TITLE
Dialogs Cancel Unexpectedly

### ISSUE THREAD

**Issue by miqrogroove** [fisharebest/webtrees#5062]
This is a follow up to #3319 to discuss if further improvements are possible.

Data loss caused by unwanted cancellation of dialogs continues to be a cause of frustration in webtrees.  There are two common scenarios:

1. The "media type" drop down list, previously described, is vulnerable to accidental double clicks.  The first click selects a type, and the second click deletes all inputs.
2. Touch screens.  It's just a fact of life that sometimes I accidentally touch the edge of the screen, which has the unfortunate effect of deleting all inputs.

Currently using webtress v2.1.20.

**Comment by fisharebest** [comment]
We fixed this in #3319 by adding `data-backdrop="static".

Change: https://github.com/fisharebest/webtrees/commit/1a1d496735f10a19072870ab8f21e5a407e1ec6c
Documentation: https://getbootstrap.com/docs/5.3/components/modal/#options

This change presumably worked at the time.

**Comment by miqrogroove** [comment]
Did that change even exist in the v2.1 branch?

**Comment by LarsRabe** [comment]
> Did that change even exist in the v2.1 branch?

I don't see a folder app/functions in 2.1 or in main? So were did the fix go?

**Comment by fisharebest** [comment]
> So were did the fix go?

We still have `data-backdrop="static"`, and the documentation says this should provide a backdrop that doesn't close the dialog when clicked.  The example on the documentation site works, so there must be something in our code that breaks this behaviour.  I just haven't found it yet.

**Comment by miqrogroove** [comment]
When I look at the page source I can't find that attribute anywhere.  Are you sure it's making it to the output?  Where's the new file?

**Comment by fisharebest** [comment]
There are about 25 occurrences of it.  e.g.
https://github.com/fisharebest/webtrees/blob/087b1e5f32b2f0d4823d27934eac6fcd8122b029/app/Elements/XrefMedia.php#L62

**Comment by fisharebest** [comment]
One of the bootstrap upgrades (4 -> 5?) added a `bs-` prefix to all its data attribute.

webtrees does similar - all ours have `wt-`.

**Comment by miqrogroove** [comment]
So yes, there is an obvious flaw here.  When I open, for example

`/tree/tree1/add-fact/I1/OBJE`

The `data-bs-backdrop` attribute has been applied to the `button` element, which is wrong.  When the XHR fires, it returns a modal `div` that never mentions the needed attribute.

**Comment by fisharebest** [comment]
Good spot!

**Comment by miqrogroove** [comment]
You probably found it already, but looks like the attribute should move into `/resources/views/modals/ajax.phtml`
